### ISSUE TITLE
Malformed filelists being generated

### ISSUE THREAD

**Issue by maksis** [eiskaltdcpp/eiskaltdcpp#359]
From http://adc.sourceforge.net/ADC.html#_file_names_and_structure :

> each file or directory name must be unique in a case-insensitive context

Eiskalt seems to be ignoring the specs: https://github.com/eiskaltdcpp/eiskaltdcpp/commit/cbc3ae53a1259b5c7a1549e7c824e0b6bc497b6d (ShareManager section)

https://bugs.launchpad.net/dcplusplus/+bug/1665900

**Comment by felisucoibi** [comment]
yes this bugs affects lot of linux users that shares dc server with windows users aind airdc, it shoud ignore case sensitive duplicates o rename them in filelists.

**Comment by klondi** [comment]
[22:51:43] [ 83.252.140.134]  <klondike> I guess it has to do with Linux having case sensitive folders unlike windows ￼
[22:52:03] [ 83.252.140.134]  <klondike> Hard to fix on Linux because then the directories would conflate.

[22:52:32] [   91.132.86.93]  <maksalaatikko> i'd say that it has more to do with your client that isn't handling those correctly
[22:52:49] [   91.132.86.93]  <maksalaatikko> airdc won't generate such filelists
[22:52:52] [   91.132.86.93]  <maksalaatikko> dc++ neither
[22:53:41] [ 83.252.140.134]  <klondike> Because it relies on case insensitive path names?
[22:54:50] [   91.132.86.93]  <maksalaatikko> yes, specifically to avoid such cases
[22:55:25] [   91.132.86.93]  <maksalaatikko> https://github.com/eiskaltdcpp/eiskaltdcpp/issues/359
[22:59:20] [ 83.252.140.134]  <klondike> I can try to write a patch then. How would you handle the case of colliding filenames?
[22:59:34] [   91.132.86.93]  <maksalaatikko> combine the directories
[22:59:52] [ 83.252.140.134]  <klondike> What if the collision happens for the files themselves?
[23:00:06] [   91.132.86.93]  <maksalaatikko> you might want to check out the code from dc++
[23:00:33] [   91.132.86.93]  <maksalaatikko> hmm, i think it adds numbers after the duplicates
[23:00:41] [ 83.252.140.134]  <klondike> Do they handle that? I thought they went for the Windows assumption that they remain the same.
[23:02:44] [ 83.252.140.134]  <klondike> I'll try to check that and write a patch then
[23:03:00] [ 83.252.140.134]  <klondike> Thanks maksalaatikko !

**Comment by klondi** [comment]
So:
1. Check how this is handled on DC++, removing folders is a bad idea so either merge or rename.
2. Fix the code that generates the xml file.

**Comment by felisucoibi** [comment]
Maybe you should remove ips from the log, anyway thanks for keep alive this bug and try to fix it.

**Comment by klondi** [comment]
IPs? Which IPs? :P

**Comment by Tantrix** [comment]
Bump!
